bigframes/core/blocks.py

```diff
@@ -487,9 +460,8 @@ def to_arrow(
             list(self.value_columns) + list(self.index_columns)
         )
 
-        _, query_job = self.session._execute(expr, ordered=ordered)
-        results_iterator = query_job.result()
-        pa_table = results_iterator.to_arrow()
```

Comment:
It seems that we should always provide the read client to avoid repeated client creation overhead - but otherwise just let the client decide when to use the read api or not? Seems the bq python client is more likely to maintain good logic for choosing the read path than we are.

Reply:
Switched to just always providing the client. It seems that to_arrow() would generate its own client if needed, but to_arrow_iterable will only use a provided client, while still using old read path where its more efficient.